File: leetcode_ly/java/Tree/KthSmallestElementInBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> list = new ArrayList<Integer>();
        getList(root,list);
        return list.get(k - 1);
    }
    
    public void getList(TreeNode root,List<Integer> list){
        if(root == null) return;
        getList(root.left,list);
        list.add(root.val);
        getList(root.right,list);
        return;
    }
}

File: leetcode_ly/c++/Tree/KthSmallestElementInBST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> ret;
    
    int kthSmallest(TreeNode* root, int k) {
        inorder(root);
        return ret[k - 1];
    }
    
    void inorder(TreeNode* root){
        if(!root) return;
        inorder(root->left);
        ret.push_back(root->val);
        inorder(root->right);
    }
};